framework/wazuh/core/cluster/dapi/dapi.py

```diff
@@ -100,6 +103,8 @@ def __init__(self, f: Callable, logger: logging.getLogger, f_kwargs: Dict = None
 
         self.local_clients = []
         self.local_client_arg = local_client_arg
+        if api_timeout:
+            api_constants.API_REQUEST_TIMEOUT = api_timeout
```

Comment:
If the timeout is different for a specific endpoint, for example in `GET /manager/status`:
```PYTHON
dapi = DistributedAPI(f=manager.get_status,
                      f_kwargs=remove_nones_to_dict(f_kwargs),
                      request_type='local_any',
                      is_async=False,
                      wait_for_complete=wait_for_complete,
                      logger=logger,
                      rbac_permissions=request['token_info']['rbac_policies'],
                      api_timeout=5
                      )
```

once the endpoint is run and the constant updated, will the timeout value be changed for all other endpoints as well, from that moment on?

Reply:
Exactly, from that moment on it would be changed for all endpoints. The explanation that this is so and the timeout is not a simple variable of the DistributedAPI class, is the line `L#394` of the `core/agent.py` file.

We cannot access from this file to the class variable of an instance of DistributedAPI, because it is this object that encompasses the execution of this code. This way right now is not enabled the possibility of having custom timeouts for some endpoints, as in version 4.1.